app/code/Magento/User/Model/User.php

```diff
@@ -347,19 +357,23 @@ protected function validatePasswordChange()
     {
         $password = $this->getPassword();
         if ($password && !$this->getForceNewPassword() && $this->getId()) {
-            $errorMessage = __('Sorry, but this password has already been used. Please create another.');
+            $errorMessage = __('Sorry, but this password is being used now. Please create another.');
             // Check if password is equal to the current one
             if ($this->_encryptor->isValidHash($password, $this->getOrigData('password'))) {
                 return [$errorMessage];
             }
 
-            // Check whether password was used before
-            foreach ($this->getResource()->getOldPasswords($this) as $oldPasswordHash) {
-                if ($this->_encryptor->isValidHash($password, $oldPasswordHash)) {
-                    return [$errorMessage];
+            $errorMessage = __('Sorry, but this password has already been used. Please create another.');
+            if ($this->observerConfig->isAllowedToRepeatPreviousPasswords() === false) {
```

Comment:
I would recommend to simplify this condition in the following way:
`if (!$this->observerConfig->isAllowedToRepeatPreviousPasswords()) {`

Reply:
I thing that "=== false" is more strict, than "!", so I don't see the reason to change it :)

Reply:
It's redandand, because `isAllowedToRepeatPreviousPasswords` always return boolean value :)